src/read.rs

```diff
@@ -209,7 +209,7 @@ fn make_reader<'a>(
 impl<R: Read + io::Seek> ZipArchive<R> {
     /// Get the directory start offset and number of files. This is done in a
     /// separate function to ease the control flow design.
-    fn get_directory_counts(
+    pub fn get_directory_counts(
```

Comment:
`pub(crate)` will do the trick!

Reply:
That's nice. BTW, I saw in another PR that you mention about the current situation of having various edge-case API sprinkled around, and I really appreciate your works in #156 in your limited time. I'll see if I can help.